### PR TITLE
Added problem badges

### PR DESCRIPTION
**Pull request by vmcj**
image

### CODE REVIEW

**Review by nickygerritsen — APPROVED** [review]
Code style warnings but besides that LGTM

File: webapp/src/Controller/Jury/ExecutableController.php [hunk 1]
```diff
@@ -59,12 +59,25 @@ public function indexAction(Request $request): Response
...
+            'badges' => ['title' => 'problems', 'sort' => false],
             'description' => ['title' => 'description', 'sort' => true,],
         ];
 
         $propertyAccessor  = PropertyAccess::createPropertyAccessor();
         $executables_table = [];
         foreach ($executables as $e) {
+            $badges = [];
+            if ($this->dj->getCurrentContest()) {
+                foreach([$e->getProblemsCompare(), $e->getProblemsRun()] as $problemList) {
```
Comment (meisterT) [inline webapp/src/Controller/Jury/ExecutableController.php:71]: If you use `array_merge` here, you can reduce the two outer loops to a single one

File: webapp/src/Controller/Jury/ExecutableController.php [hunk 2]
```diff
@@ -91,6 +104,7 @@ public function indexAction(Request $request): Response
                 default:
                     $execdata['icon']['icon'] = 'question';
             }
+            $execdata['badges']['value'] = $badges;
```
Comment (meisterT) [inline webapp/src/Controller/Jury/ExecutableController.php:107]: Should we add an explicit sort here?

**Comment by vmcj** [comment]
image

I tried to improve the speed of the queries when handling Tobi his remarks. @nickygerritsen this is non-trivial with Doctrine is there a better way?

File: webapp/src/Controller/Jury/ExecutableController.php (hunk 1 above)
Comment (vmcj) [reply webapp/src/Controller/Jury/ExecutableController.php]: Tried to fix this in the query to speed it up further.

File: webapp/templates/jury/jury_macros.twig [hunk 3]
```diff
@@ -141,6 +141,10 @@
                             {{- (item.value|default(item.default|default('')))|affiliationLogo(item.title) -}}
                             {% elseif key == "warning_content" %}
                             {{- item.value|printWarningContent -}}
+                            {% elseif key == "badges" %}
+                                {% for badge in item.value %}
+                                    {{- badge|problemBadge -}}
```
Comment (nickygerritsen) [inline webapp/templates/jury/jury_macros.twig:146]: Maybe we want a space between the badges?

**Review by nickygerritsen — APPROVED** [review]
> I tried to improve the speed of the queries when handling Tobi his remarks. @nickygerritsen this is non-trivial with Doctrine is there a better way?

Besides doing direct queries or maybe a DQL, I don't think so

File: webapp/templates/jury/jury_macros.twig (hunk 3 above)
Comment (vmcj) [reply webapp/templates/jury/jury_macros.twig:146]: image

Fine by me.

**Review by meisterT — APPROVED** [review]

**Pull request merged**